### PR TITLE
Add green task form to auth patterns form app

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Builds out the 'green task' for authenticated patterns mock form. This form is only for staging and will not be launched to production
- Adds introduction, veteran info, mailing address, and edit address pages that are similar to the 1010EZR form
- Task buttons to navigate between the task versions are located at the bottom of the screen
- The beginning of the yellow task config has been started, but is mostly just a starter template that was used to verify that the task buttons worked. The purple task has not been started.
- Lots of boilerplate code copied over from the 1010ezr to establish this first mock form, utils, and alerts
- useLocalStorage hook is used to help mock auth. This is still a WIP but works locally as expected.
- redux store established and an action to set a new form config was created so that each task could start a whole new form flow when needed. We use the url structure to determine what config to load.

I realize that this is a massive PR, mostly because we had to copy a lot of code over from the EZR form to get this first task bootstrapped, so let me know if anything seems out of place or could be cleaned up.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/tmf-auth-exp-design-patterns/issues/51

## Testing done

- Tested locally, this application is a mock form and is not being launched to prod

### Steps to test locally:

- pull in branch, make sure the yarn install is recent
-  run the FE in watch mode with `yarn watch --env entry=mock-form-ae-design-patterns`
- run the mock server with `yarn mock-api --responses src/applications/_mock-form-ae-design-patterns/mocks/server.js`
- open the green task at url `http://localhost:3001/mock-form-ae-design-patterns/task-green` and verify that the form loads

## Screenshots

These are all new pages, so we didnt provide before screenshots

Intro page

page-1

---

Page 2 - Uneditable info

screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-personal-information-2024-08-12-14_14_16

---

Page 3 - Editable info for mailing address

screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-confirm-mailing-address-2024-08-12-14_14_27

---

Page 4 - Editing Mailing Address

screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-veteran-information-edit-mailing-address-2024-08-12-14_14_46

---

Page 5 - Task complete (this is just an idea I had)

screencapture-localhost-3001-mock-form-ae-design-patterns-task-green-complete-2024-08-12-14_15_00

## What areas of the site does it impact?

Doesn't impact production site. On staging it impacts the /mock-form-ae-design-patterns url

## Acceptance criteria

- Pages built from figma designs
- Task buttons load up a different task (just yellow and green set up so far)

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by msbtterswrth — APPROVED** [review]
LGTM, ship it :)

File: src/applications/_mock-form-ae-design-patterns/components/FormPages/VeteranProfileInformation.jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import moment from 'moment';
```
Comment (jamigibbs) [inline src/applications/_mock-form-ae-design-patterns/components/FormPages/VeteranProfileInformation.jsx:4]: If possible can this be updated at some point to use `date-fns`? 

https://dsva.slack.com/archives/C5HP4GN3F/p1721065353985949

**Review by jamigibbs — APPROVED** [review]
Go forth and patternize 🫡

**Review by asiisii — APPROVED** [review]

**Review by bellepx0 — APPROVED** [review]
Awesome work! Approved

**Pull request merged**